### DISCUSSION TITLE
Naming for NiceGUI's new React-like API

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/nicegui#1489] [Polls]
In PRs #1438 and #1447 we're developing a new API for writing reactive UI functions that behave similar to React:

```py
@ui.reactive
def counter():
    count, set_count = ui.use_state(0)
    ui.label(f'Count: {count}')
    ui.button('Increment').on('click', lambda: set_count(count + 1))

counter()
```

When using `set_count()` inside the click handler, the UI is automatically updated.

In PR #1447 we found a rather lightweight implementation based on `ui.refreshable`, so that you can also call `counter.refresh` programmatically. From this point of view `ui.reactive` and `ui.refreshable` are very similar, suggesting to use the name `ui.refreshable` even for reactive functions with `use_state()`. But this might confuse React users and doesn't perfectly indicate the reactive behavior.

So my question is: How should the API for auto-refreshing UI functions be?

1. `ui.refreshable`, no react
    We would keep NiceGUI as it is and don't add the new reactive behavior, basically to keep things simple.
2. `ui.refreshable`, `ui.state`
    We extend the existing `ui.refreshable` to support the new `ui.state` (without the `use_` prefix from the example above).
3. `ui.xxx`, `ui.yyy` (option 2 with different names)
    We extend the existing `ui.refreshable` to support "UI state", but give different names. Please comment with your suggestion.
4. `ui.refreshable` + `ui.reactive`/`ui.use_state`
    We keep `ui.refreshable` unchanged and introduce reactive functions named `ui.reactive` and `ui.use_state` to stay close to React.

What do you think? Do you have additional thoughts and ideas?

**Comment by thickmn** [comment]
I think option 4 is most intuitive when combined with the team's format of documentation (with an example for clarity/practicability).

**Comment by Januswork** [comment]
Option 2 simple, easy to use

**Comment by falkoschindler** [comment]
Ok, thanks everyone for taking part in this survey!
Although there are about 40% voting for a more React-like naming scheme, the majority of 60% votes for `ui.refreshable` and `ui.state`.
I'm going to close this poll and implement the result in PR #1447. 👍🏻
